video/out/wayland_common.c

```diff
@@ -1297,3 +1299,18 @@ bool vo_wayland_config (struct vo *vo, uint32_t flags)
 
     return true;
 }
+
+void vo_wayland_request_frame(struct vo *vo, void *data, vo_wayland_frame_cb cb)
+{
+    struct vo_wayland_state *wl = vo->wayland;
+    wl->frame.data = data;
+    wl->frame.function = cb;
+    MP_DBG(wl, "restart frame callback\n");
+    frame_callback(wl, NULL, 0);
+}
+
+void vo_wayland_wait_frame(struct vo *vo)
+{
+    struct vo_wayland_state *wl = vo->wayland;
+    wl_display_roundtrip(wl->display.display);
```

Comment:
This looks like it'd also return after new mouse events and such. Wouldn't this lead to bad results? (But I don't know what this function does.)